classes/class-wc-klarna-order-management-scheduled-actions.php

```diff
@@ -13,48 +13,47 @@
 }
 
 /**
- * WC_Klarna_Order_Actions_Display class.
+ * WC_Klarna_Order_Management_Display_Actions class.
  *
  * Displays scheduled actions related to the order.
  */
-class WC_Klarna_Order_Actions_Display {
+class WC_Klarna_Order_Management_Display_Actions {
 
 	/**
-	 * Retrieves and displays scheduled actions for the order.
+	 * Gets the scheduled actions for the order.
 	 *
 	 * @param string $session_id The session ID.
-	 * @return void
+	 * @return array
 	 */
-	public static function get_scheduled_actions( $session_id ) {
-		$session_query_url = admin_url(
-			'admin.php?page=wc-status&tab=action-scheduler&s=' . rawurlencode( $session_id ) . '&action=-1&paged=1&action2=-1'
-		);
-
-		$statuses      = array( 'complete', 'failed', 'pending' );
-		$action_counts = array();
+	private static function get_scheduled_actions( $session_id ) {
+		$statuses          = array( 'complete', 'failed', 'pending' );
+		$scheduled_actions = array();
 
 		foreach ( $statuses as $status ) {
-			$action_counts[ $status ] = count(
+			$scheduled_actions[ $status ] =
 				as_get_scheduled_actions(
 					array(
 						'search'   => $session_id,
 						'status'   => array( $status ),
 						'per_page' => -1,
 					)
-				)
-			);
+				);
```

Comment:
I've encounter and issue with making a query like this recently as well, not towards the scheduled actions table granted, but towards the order table. But it might be a good idea to test and see how bad the performance is for this before we make a release with this. Just to verify that this does not cause a memory or execution time issue. Don't make any changes as of now, we can check this together though just to make sure.